Source/WorldServer/Server.cs

```diff
@@ -190,5 +204,80 @@ static void ExitNow()
             Thread.Sleep(5000);
             Environment.Exit(-1);
         }
+
+        static async void StartDiscordBot()
+        {
+            DiscordSocketClient client = new DiscordSocketClient();
+
+            client.Log += LogDiscord;
+            client.MessageReceived += SendToDiscord;
+
+            string token = ConfigMgr.GetDefaultValue("Discord.BotToken", "");
+
+            if (token != "")
+            {
+                Console.WriteLine($"Starting bot with token: {token}");
+
+                await client.LoginAsync(TokenType.Bot, token);
+                await client.StartAsync();
+
+                while (true)
+                {
+                    if (!DiscordMessageQueue.Empty())
+                    {
+                        foreach (DiscordMessage message in DiscordMessageQueue.DiscordMessages)
+                        {
+                            string formatedMessage = "";
+                            formatedMessage = GetFormatedMessage( message );
+
+                            ulong serverID = ConfigMgr.GetDefaultValue("Discord.BotServerID", default(ulong));
+                            ulong channelID = default( ulong );
+
+                            if (serverID != default( ulong ))
+                            {
+                                switch (message.Channel)
+                                {
+                                    case DiscordMessageChannel.Discord_Both:
+                                        channelID = ConfigMgr.GetDefaultValue("Discord.BotChannelIDBoth", default(ulong));
+                                        break;
+                                    case DiscordMessageChannel.Discord_World_A:
+                                        channelID = ConfigMgr.GetDefaultValue("Discord.BotChannelIDAlliance", default(ulong));
+                                        break;
+                                    case DiscordMessageChannel.Discord_World_H:
+                                        channelID = ConfigMgr.GetDefaultValue("Discord.BotChannelIDHorde", default(ulong));
+                                        break;
+                                }
+                                if( channelID != default(ulong))
+                                    await client.GetGuild(serverID).GetTextChannel(channelID).SendMessageAsync(formatedMessage);
+                            }
+                        }
+                        DiscordMessageQueue.Clear();
+                    }
+                    Thread.Sleep(2000);
+                }
+                await Task.Delay(-1);
+            }
+        }
+
+        private static string GetFormatedMessage(DiscordMessage message)
+        {
+            string blizzIcon = message.IsGm ? "<:blizz:407908963007594496>" : "";
```

Comment:
Emojis like this are only usable by bots when they are in a server which has access to the emoji - you'll probably want to move this to an external configuration file if possible.

Reply:
This what? The emoji icon file or...?